tensorflow/compiler/xla/literal_util.cc

```diff
@@ -917,6 +933,16 @@ LiteralUtil::GetMutableArraySlice(Literal* literal) {
 }
 
 template <>
+/* static */ tensorflow::gtl::MutableArraySlice<half>
```

Comment:
Literals are certainly stored on disk. I don't know whether you have discovered them, but you can dump computations to disk (TF_XLA_FLAGS=--xla_dump_computations_to=/tmp/somewhere) and there are utilities to visualize and replay dumped computations in tensorflow/compiler/xla/tools. So the potential for the problem is there, at least --- you can copy the dumped computations between machines.

I agree that it would be nice if protobuf added a 16-bit type, but it won't happen immediately even if they agree.